### PR TITLE
installer: autorun xserver in graphical media

### PR DESCRIPTION
**Pull request by worldofpeace**
###### Motivation for this change
When I first used the NixOS iso I found it confusing and strange to have to do `systemctl start display-manager`. I assumed that I would have went straight to the graphical environment.

If anyone has a reason as to why it shouldn't do this please share.

###### Notes
In #42610 @adisbladis suggested:
> I think we should add another boot entry so you can still choose not to start the graphical environment, at the same time we make the one that autostarts the graphical environment the default.

Wouldn't this need https://github.com/NixOS/nixpkgs/pull/45345 to include that configuration as a child config in the grub menu?

Though I feel that might be to complex.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---
cc @samueldr @xeji @grahamc

### CODE REVIEW

**Comment by samueldr** [comment]
I would like to know why it wasn't the default. There is probably a reason behind, and knowing it helps ensuring we don't cause any unintended issues.

* * *

Though, I think this requires one important sub-feature: the installer media should present an option that disables the `display-manager.service` so one can use the graphical iso without having X started; in case starting X is causing issues. IIRC this can be done through a [`systemd.mask=display-manager.service`](https://www.freedesktop.org/software/systemd/man/kernel-command-line.html) option on the kernel command line.

**Comment by samueldr** [comment]
Yes, sorry, bad habit, jumped the gun and added my thoughts before reading that this was already in the body of your PR.

This is one way the boot option can be implemented.

**Comment by worldofpeace** [comment]
No problem, I'm guilty of the same thing too when if I was busy.

**Comment by worldofpeace** [comment]
If there's an agreement to needing this feature implicitly then I'd say that this is blocked until
the tests for https://github.com/NixOS/nixpkgs/pull/45345 are reviewed and it's merged.

**Comment by samueldr** [comment]
Probably no relation. The installer iso has a different menu generation entirely from the installed system.

[Around here](https://github.com/NixOS/nixpkgs/blob/aec67d299c24d0bfff6f5ae04b5c462f3d3138d9/nixos/modules/installer/cd-dvd/iso-image.nix#L49-L65) a new option can be added (class is used to get an icon in the menu).

**Comment by mmahut** [comment]
Any updates on this pull request, please?

**Comment by worldofpeace** [comment]
I plan to pick this up again after the 19.09 release.

**Comment by worldofpeace** [comment]
Ok, rebased and added a quirk to disable the display-manager.
I think we might've discussed on having a way to condition menu entries on whether it's for the graphical media or minimal. Though I still think this change won't do much harm without that feature.

**Comment by worldofpeace** [comment]
I've tested running the graphical media
* Does it autorun X starting plasma5 :heavy_check_mark: 
* Did enabling the quirk send to me to the tty :heavy_check_mark:

File: nixos/modules/installer/cd-dvd/iso-image.nix
Comment (samueldr) [inline nixos/modules/installer/cd-dvd/iso-image.nix]: (it's not an issue per se) should it be autoran? or rephrased? English is weird, even weirderer when neologisms are introduced! ;)

File: nixos/modules/installer/cd-dvd/iso-image.nix
Comment (samueldr) [inline nixos/modules/installer/cd-dvd/iso-image.nix]: We can introduce an icon, if desired, and if one can be composed. They are composed from the [breeze icon theme](https://github.com/KDE/breeze-icons). If you see something in it that can be adapted, or used as an inspiration I can look into [adding it to the relevant artwork](https://github.com/NixOS/nixos-artwork/tree/master/bootloader/grub2-installer).

File: nixos/doc/manual/release-notes/rl-2003.xml
Comment (samueldr) [inline nixos/doc/manual/release-notes/rl-2003.xml]:
```suggestion
      Before you'd be greeted by a tty and asked to enter <command>systemctl start display-manager</command>.
```

File: nixos/doc/manual/release-notes/rl-2003.xml
Comment (samueldr) [inline nixos/doc/manual/release-notes/rl-2003.xml]:
```suggestion
      It is now possible to disable the display-manager from running by selecting the <literal>Disable display-manager</literal>
```

File: nixos/doc/manual/release-notes/rl-2003.xml
Comment (samueldr) [inline nixos/doc/manual/release-notes/rl-2003.xml]:
```suggestion
      quirk in the boot menu.
```

GRUB is an implementation detail; though unlikely some users may not know what a GRUB is.

**Review by samueldr — CHANGES_REQUESTED** [review]
Minor stuff, generally approving the change.

**Comment by worldofpeace** [comment]
@samueldr My English and personality is literally a neologism :rofl: 

Thanks for the suggestions, will try to implement.

File: nixos/modules/installer/cd-dvd/iso-image.nix
Comment (worldofpeace) [reply nixos/modules/installer/cd-dvd/iso-image.nix]: I found breeze/devices/64/video-display.svg the most suitable.
I mostly wanted something like just a blank monitor, or communicates a tty.

Comment (worldofpeace) [reply nixos/modules/installer/cd-dvd/iso-image.nix]: I'm highly partial to elementary's icons which appears to be exactly what I'm looking for

Screenshot from 2019-09-28 21 14 48

File: nixos/modules/installer/cd-dvd/iso-image.nix
Comment (worldofpeace) [reply nixos/modules/installer/cd-dvd/iso-image.nix]: I believe it's autoran, but it appears I've used multiple alternates even in the commit messages.

File: nixos/modules/installer/cd-dvd/iso-image.nix
Comment (samueldr) [reply nixos/modules/installer/cd-dvd/iso-image.nix]: Oh, forgot to specify the symbolic icons. Though the one from breeze, once made into outlines, may work, I'll try to remember to look into it later. (The PR can be merged without that change.)

Comment (worldofpeace) [reply nixos/modules/installer/cd-dvd/iso-image.nix]: Ahh, there's computer/video-display symbolic icons that should be fine then.
I'll add an issue at nixos-artwork for that once merged @samueldr.

**Comment by worldofpeace** [comment]
Rebased and resolved the conflicts in the release note.
I plan to integrate this some time today if there's no issues raised.

**Pull request merged**
